core/src/main/java/ai/timefold/solver/core/impl/heuristic/selector/move/generic/RuinRecreateConstructionHeuristicPhase.java

```diff
@@ -28,4 +40,26 @@ public String getPhaseTypeString() {
         return "Ruin & Recreate Construction Heuristics";
     }
 
+    @Override
+    protected void doStep(ConstructionHeuristicStepScope<Solution_> stepScope) {
+        if (elementsToRuinSet != null) {
+            var listVariableDescriptor = stepScope.getPhaseScope().getSolverScope().getSolutionDescriptor()
+                    .getListVariableDescriptor();
+            var entity = stepScope.getStep().extractPlanningEntities().iterator().next();
+            if (!elementsToRuinSet.contains(entity)) {
+                // Sometimes, the list of elements to be ruined does not include new destinations select by the CH.
+                // In these cases,
+                // we need to record the element list before making any move changes
+                // so that it can be referenced
+                // to restore the solution to its original state in the inner nodes.
```

Comment:
```suggestion
                // Sometimes, the list of elements to be ruined does not include new destinations selected by the CH.
                // In these cases, we need to record the element list before making any move changes
                // so that it can be referenced to restore the solution to its original state in the inner nodes.
```

Reply:
What nodes are we talking about here? Perhaps a different word is in order? "Node" is a deep CS terminology.

Reply:
I meant the score director restoring the original state.